maskrcnn_benchmark/modeling/roi_heads/roi_heads.py

```diff
@@ -33,26 +34,27 @@ def forward(self, features, proposals, targets=None):
                 self.training
                 and self.cfg.MODEL.ROI_MASK_HEAD.SHARE_BOX_FEATURE_EXTRACTOR
             ):
-                mask_features = x
-            # During training, self.box() will return the unaltered proposals as "detections"
+                mask_features = box_features
+            # During training, self.mask() will return the unaltered proposals as "detections"
             # this makes the API consistent during training and testing
-            x, detections, loss_mask = self.mask(mask_features, detections, targets)
+            _, detections, loss_mask = self.mask(mask_features, detections, targets)
             losses.update(loss_mask)
 
         if self.cfg.MODEL.KEYPOINT_ON:
             keypoint_features = features
             # optimization: during training, if we share the feature extractor between
-            # the box and the mask heads, then we can reuse the features already computed
+            # the box and the keypoint heads, then we can reuse the features already computed
             if (
                 self.training
                 and self.cfg.MODEL.ROI_KEYPOINT_HEAD.SHARE_BOX_FEATURE_EXTRACTOR
             ):
-                keypoint_features = x
-            # During training, self.box() will return the unaltered proposals as "detections"
+                keypoint_features = box_features
+            # During training, self.keypoint() will return the unaltered proposals as "detections"
             # this makes the API consistent during training and testing
-            x, detections, loss_keypoint = self.keypoint(keypoint_features, detections, targets)
+            _, detections, loss_keypoint = self.keypoint(keypoint_features, detections, targets)
             losses.update(loss_keypoint)
-        return x, detections, losses
+
+        return box_features, detections, losses
```

Comment:
This is still a change in behavior: you are now only returning the `box_features`, while before we were returning whatever was the output of the last model executed.

While I agree that it might be  good thing to do, it's not good that we do it in the middle of other changes that are completely unrelated.

Please make different PRs for different aspects.
- One PR for renaming each one of the variables (and potentially removing unused variables), but keeping the functionality **intact**
- Open one issue to discuss about potentially changing some behavior.

Mixing both changes in a single PR makes it **much harder** to review it.

Reply:
I've revert this change to the original version and I'll open another issue to discuss about it.
Changes about typos in #936 are also merged in this PR and I'll close #936 later.